src/wallet/wallet.h

```diff
@@ -729,6 +729,8 @@ class CWallet final : public WalletStorage, public interfaces::Chain::Notificati
     /** Absolute maximum transaction fee (in satoshis) used by default for the wallet */
     CAmount m_default_max_tx_fee{DEFAULT_TRANSACTION_MAXFEE};
 
+    /** Maximum transaction fee rate (in satoshis) used by default for the wallet */
+    CAmount m_default_max_tx_fee_rate{node::DEFAULT_MAX_RAW_TX_FEE_RATE.GetFeePerK()};
```

Comment:
Done, `CFeeRate` is the right format since we are representing fee rate.

Reply:
>Shouldn't there be a check in CreateTransactionInternal that makes sure we don't make transactions above this feerate, and an arg passed to BroadcastTransaction? I also don't see a test case for it.

Agreed

I Added a check in `CreateTransactionInternal` in 0c9e0c5b2218c154b62275794b878af7e1457647. 

No need to pass an arg to `BroadcastTransaction` if the transaction fee rate is above `maxfeerate` it will not read `BroadcastTransaction` so it will just be a dead code.

I added a test for it.

Reply:
Interestingly, in https://github.com/bitcoin/bitcoin/blob/411ba32af21a56efa0a570b6aa8bf8f035410230/src/rpc/mempool.cpp#L89

`sendrawtransaction` takes a fee_rate argument, which is used to calculate a `max_tx_fee` based on the size of the transaction, and this `max_tx_fee` is passed to `BroadcastTransaction`. This means a user could typo an insane `maxfeerate` when calling `sendrawtransaction` that bypasses the default max tx fee.

Perhaps it does make more sense to have `BroadcastTransaction` take two arguments, one for `max_tx_fee` and another for `max_tx_fee_rate`, and check them independently. Probably out of scope for this PR, since this is focused on the wallet, but wanted to mention it as a potential follow-up.

Reply:
Thanks for highlighting this issue @josibake , I agree with your `BroadcastTransaction` should enforce check on both.
This will make `sendrawtransaction` returns a specific error message.


---
For this reason
`MAX_FEE_RATE_EXCEEDED` error string is now `Fee exceeds maximum configured by user.`
because if I add the option `maxtxfee` it will be misleading for in `sendrawtransaction` output

Reply:
Im Open For Business